pkg/grafana/dashboard-handler.go

```diff
@@ -90,9 +79,25 @@ func (h *DashboardHandler) Unprepare(resource grizzly.Resource) *grizzly.Resourc
 
 // Prepare gets a resource ready for dispatch to the remote endpoint
 func (h *DashboardHandler) Prepare(existing, resource grizzly.Resource) *grizzly.Resource {
+	uid, _ := resource.GetSpecString("uid")
+	if uid == "" {
+		resource.SetSpecString("uid", resource.Name())
+	}
 	return &resource
 }
 
+// Validate returns the uid of resource
+func (h *DashboardHandler) Validate(resource grizzly.Resource) error {
+	uid, exist := resource.GetSpecString("uid")
+	if resource.Name() == "" {
+		return fmt.Errorf("name cannot be blank")
+	}
```

Comment:
Isn't that a base condition for all resources?

Reply:
Yeah. I did wonder while typing it. Hmm.

Reply:
I've updated the PR to do some validation for every resource before calling `handler.Validate()`.